common/writeThoughFile.go

```diff
@@ -44,19 +45,26 @@ var RootShareRegex = regexp.MustCompile(`(^\/\/[^\/]*\/?$)`)
 
 func isRootPath(s string) bool {
 	shortParentDir := strings.ReplaceAll(ToShortPath(s), OS_PATH_SEPARATOR, AZCOPY_PATH_SEPARATOR_STRING)
-	return  RootDriveRegex.MatchString(shortParentDir)  ||
-		RootShareRegex.MatchString(shortParentDir)  ||
+	return RootDriveRegex.MatchString(shortParentDir) ||
+		RootShareRegex.MatchString(shortParentDir) ||
 		strings.EqualFold(shortParentDir, "/")
 }
 
-
 func CreateParentDirectoryIfNotExist(destinationPath string, tracker FolderCreationTracker) error {
 	// If we're pointing at the root of a drive, don't try because it won't work.
 	if isRootPath(destinationPath) {
 		return nil
 	}
 
-	lastIndex := strings.LastIndex(destinationPath, DeterminePathSeparator(destinationPath))
+	pathSeparator := DeterminePathSeparator(destinationPath)
+	lastIndex := strings.LastIndex(destinationPath, pathSeparator)
+
+	// LastIndex() will return -1 if path separator was not found, we should handle this gracefully
+	// instead of allowing AzCopy to crash with an out-of-bounds error.
+	if lastIndex == -1 {
+		return fmt.Errorf("error: Path separator (%s) not found in destination path. On Linux, this may occur if the destination is the root file, such as '/'. If this is the case, please consider changing your destination path.", pathSeparator)
```

Comment:
If someone wants to copy at "/" shouldn't we allow it. Anyway if there are access issues, creation of file will fail with EPERM.

Reply:
This is a good point, but should we consider this a feature request? We can add it to our backlog for future consideration.

Currently we panic on copying to / so this improves the experience in the meantime.